L2 Transaction Fee API  specification

Therecanbeonlyone1969 opened the pull request:
Testability requirements will be added once all requirements have been agreed upon.

Therecanbeonlyone1969 committed:
Initial specification

- API 3.0 api spec
- Requirements per API

Therecanbeonlyone1969 committed:
Meeting minutes from March 6 WG meeting

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Edits based WG discussion feedback

- to proprety is not required in transactionCall
- updated currency property to reflect ability to specify native tokens and fiat currencies based on operator choice/offering.

Therecanbeonlyone1969 committed:
Removed EEA Logo and EEA -> Ethereum where warranted

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Minor editorial updates based on the L2 WG review on April 3 2024

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ethereum-oasis-op/L2?pullRequest=53) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ethereum-oasis-op/L2?pullRequest=53) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ethereum-oasis-op/L2?pullRequest=53) it.</sub>

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
L2 Transaction Fee API Testability Criteria

Added testability criteria for R1 - R5 (all API requirements).

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Editorial Updates: L2 Transaction Fee API after April 17 2024 WG discussion

- clarified the scope of the API with regard to Meta Transactions and User Actions in account abstraction
- fixed some typos in the data schema.

lgingerich commented:
I approve this draft

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Update to L2 Transaction Fee API spec

- Added definition of transaction status
- Added minimal set of transaction statuses.
- Updated requirement on `transactionStatus` data property (R4) with minimal ste of transaction statuses.
- Added definitions of minimal set of L2 transaction statuses
- changed wording on `gasPrice` requirement from number to integer

Therecanbeonlyone1969 committed:
Fixed display issue (all bold faced) in table of content

realcodywburns reviewed (commented):
no objections or comments. Looks good to me.

Therecanbeonlyone1969 committed:
Update author list

Therecanbeonlyone1969 committed:
Removed transaction statuses and their definitions and referred to specific L2 client implementations

We will revisit the status naming in a separate standard on L2 transaction status nomenclature.

Previous Definitions:
We define the minimal set of transaction statuses as follows:

- Pending: An L2 transaction submitted to an L2, and waiting to be processed
- Replaced: An L2 transaction that was "Pending" was replaced by another L2 transaction
- Dropped: An L2 transaction that was removed from the L2 processing queue
- Success: A Pending L2 transaction has been included into an L2 block
- Finalized: An L2 transaction included in a L2 block that has been finalized on an L1.

Note, that the first four statuses are analogous to L1 transaction statuses. Note also, that there is still the possibility of an L1 reorganization that could revert the status of an L2 transaction back to success. Also at the time of writing, there is no possibility of an L2 reorganization unless through a malicious attacker.

Finally, more fine-grained statuses such as "Pending L1 Finalization" could be introduced to deliver more insights to end-user and improve the user experience.

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 commented:
@smartcontracts @DZGoldman @t0mcr8se @0x000hx @leeederek -- please, review and approve this PR ASAP. Thank you!

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
Merge branch 'main' into patch-7-l2-transaction-fee-api-spec

Therecanbeonlyone1969 committed:
PR Update based on 11/27/2024 WG Meeting

- Adding to out of scope statement
- Update L1 Gas Price description to account for the price being either for call data or blob data
